src/lib.rs

```diff
@@ -139,3 +150,44 @@ impl<'a, F: Field> LabeledPolynomial<'a, F> {
         self.hiding_bound
     }
 }
+
+/// A commitment along with information about its degree bound (if any).
+#[derive(Clone)]
+pub struct LabeledCommitment<C: PCCommitment> {
+    label: String,
+    commitment: C,
+    degree_bound: Option<usize>,
+}
+
+impl<C: PCCommitment> LabeledCommitment<C> {
+    /// Instantiate a new polynomial_context.
+    pub fn new(label: String, commitment: C, degree_bound: Option<usize>) -> Self {
+        Self {
+            label,
+            commitment,
+            degree_bound,
+        }
+    }
+
+    /// Return the label for `self`.
+    pub fn label(&self) -> &String {
+        &self.label
+    }
+
+    /// Retrieve the polynomial from `self`.
+    pub fn commitment(&self) -> &C {
+        &self.commitment
+    }
+
+    /// Retrieve the degree bound in `self`.
+    pub fn degree_bound(&self) -> Option<usize> {
+        self.degree_bound
+    }
+}
+
+impl<C: PCCommitment> algebra::ToBytes for LabeledCommitment<C> {
+    #[inline]
+    fn write<W: std::io::Write>(&self, writer: W) -> std::io::Result<()> {
+        self.commitment.write(writer)
+    }
+}
```

Comment:
Should we also write the label and degree bound to bytes? We primarily use these labeled commitments as input to a hash function.